### PR TITLE
Add ensure_enqueued() fixer

### PR DESCRIPTION
**Pull request by akx**
Add a fixer function that can make sure an unprocessed job is actually in the queue.

### CODE REVIEW

File: minique_tests/test_minique.py [hunk 1]
```diff
@@ -46,3 +46,17 @@ def test_cancel(redis: Redis, random_queue_name: str) -> None:
...
+
+
+def test_ensure_enqueued(redis: Redis, random_queue_name: str) -> None:
+    job = enqueue(redis, random_queue_name, 'minique_tests.jobs.sum_positive_values')
+    queue = job.get_queue()
+    assert queue.length == 1
+    assert job.ensure_enqueued() == (False, 0)  # Did not need to re-enqueue
+    queue.clear()
+    assert queue.length == 0
+    assert job.ensure_enqueued() == (True, 0)  # Did re-enqueue
```
Comment (ruksi) [inline minique_tests/test_minique.py:58]: nitpick; would've tested with two items and specifically removing one of them to check that 1) the index works, now it's not actually testing other than "it's zero" and 2) it's only touching stuff that it should be re-enqueueing when it does a re-enqueue but based on the current implementation logic, seems solid

**Review by ruksi — APPROVED** [review]
lgtm, some nitpicks that can be rectified or not

File: minique_tests/test_minique.py (hunk 1 above)
Comment (akx) [reply minique_tests/test_minique.py:58]: 👍, can definitely embetter the test

File: minique_tests/test_minique.py [hunk 2]
```diff
@@ -46,3 +46,23 @@ def test_cancel(redis: Redis, random_queue_name: str) -> None:
...
+    assert j1.ensure_enqueued() == (False, 0)  # Did not need to re-enqueue
+    assert j2.ensure_enqueued() == (False, 1)  # Did not need to re-enqueue
+    assert redis.lpop(queue.redis_key) == j1.id.encode()  # pop first item, must be the first job
+    assert queue.length == 1
+    assert j1.ensure_enqueued() == (True, 1)  # Did re-enqueue in last position
+    assert j2.ensure_enqueued() == (False, 0)  # Did not need to re-enqueue
+    Worker.for_queue_names(redis, queue.name).tick()
+    assert queue.length == 1
+    Worker.for_queue_names(redis, queue.name).tick()
+    assert queue.length == 0
```
Comment (ruksi) [inline minique_tests/test_minique.py:65]: 👌

**Pull request merged**